### PR TITLE
Tickets/DM-44529: fix generateDonutDirectDetect for null selection

### PR DESCRIPTION
**Pull request by suberlak**
This PR adds a fix to the direct detection task, whereby if sources are detected, but none selected via `donutSelector`, the task would produce an error failing entire pipeline. Now it will return a `warning` similar to the case of no donuts detected, alongside with an empty donut table.

### CODE REVIEW

File: python/lsst/ts/wep/task/generateDonutDirectDetectTask.py [hunk 1]
```diff
@@ -273,27 +300,26 @@ def run(self, exposure, camera):
...
-                [detectorName] * len(donutCatUpd), dtype=str
-            )
+            # If at least one donut got selected, update the column names
+            # and content
+            if len(donutCatSelected) > 0:
+                donutCatUpd = self.updateDonutCatalog(donutCatSelected, exposure)
+                donutCatUpd["detector"] = np.array(
+                    [detectorName] * len(donutCatUpd), dtype=str
+                )
+            # If no donuts got selected, issue a warning and return an enpy
```
Comment (jbkalmbach) [inline python/lsst/ts/wep/task/generateDonutDirectDetectTask.py]: typo "enpy"

File: tests/task/test_generateDonutDirectDetectTask.py [hunk 2]
```diff
@@ -236,6 +269,39 @@ def testTaskRun(self):
...
+            expected_metakeys,
+        )
+
+        # Test the behavior when source selection is turned off
+        self.task.config.doDonutSelection = False
+        taskOut_S11_noSelection = self.task.run(
+            exposure_S11,
+            camera,
+        )
+        # Check that the expected columns are present
```
Comment (jbkalmbach) [inline tests/task/test_generateDonutDirectDetectTask.py:300]: Can you add a test to make sure there are sources when you do this just to check that it's not an empty catalog?

**Review by jbkalmbach — APPROVED** [review]
Looks good. Minor comments. Thanks for fixing this!

File: python/lsst/ts/wep/task/cutOutDonutsBase.py [hunk 3]
```diff
@@ -611,36 +611,38 @@ def cutOutStamps(self, exposure, donutCatalog, defocalType, cameraName):
...
 
             # Prepare blend centroid position information
-            if len(catalogMeta["blend_centroid_x"][idx]) > 0:
-                blendCentroidPositions = np.array(
-                    [
-                        catalogMeta["blend_centroid_x"][idx] + donutRow["xShift"],
-                        catalogMeta["blend_centroid_y"][idx] + donutRow["yShift"],
-                    ]
-                ).T
+            if len(catalogMeta["blend_centroid_x"]) > 0:
```
Comment (jbkalmbach) [inline python/lsst/ts/wep/task/cutOutDonutsBase.py]: Can you combine the logic here so you don't have the same command under two nested else statements?

Comment (jbkalmbach) [reply python/lsst/ts/wep/task/cutOutDonutsBase.py]: Actually a better solution to all of this would be to check before the `for` loop starts on line 583 and:
```
if "blend_centroid_x" not in catalogMeta.keys(): # Or whatever best way to check there's no blend data
    catalogMeta["blend_centroid_x"] = []*len(donutCatalog)
    catalogMeta["blend_centroid_y"] = []*len(donutCatalog)
```
that way you won't have to add nested logic inside each iteration of the `for` loop.

Comment (suberlak) [reply python/lsst/ts/wep/task/cutOutDonutsBase.py]: Ok, I implemented a different way that's more straightforward. We make the output of null selection (`donutSelector` was on , but no sources selected) same as  no selection (`donutSelector` was off, so that property was not calculated):  `donutCatalog.meta['blend_centroid_x'] = ""` 
Then in `cutOutDonutsBase` we similarly only use the `blend_centroid_x` if it is longer than 0. 

Also, I noticed that `finalBlendXList` gets created and updated, but not persisted anywhere (not in the `DonutStamp` nor in `stampsMetadata`. Shall I add it to the metadata, or remove?
